frontend/src/Components/ContentComponent.tsx

```diff
@@ -225,14 +228,25 @@ export default function ContentComponent(props: ContentComponentProps) {
             return;
         }
 
-        createObserverService();
         updateContent(content);
 
-        if (props.autoCut) {
-            const rect = content.getBoundingClientRect();
-            if (rect.height > 1000) {
-                setCut(true);
+        let resizeObserver: ResizeObserver | null = null;
+        const handleResize = (entries: ResizeObserverEntry[]) => {
+            for (const entry of entries) {
+                if (props.autoCut && entry.target === content) {
+                    const rect = entry.contentRect;
+                    if (rect.height > props.autoCut + 250) {
+                        setCut(true);
+                        resizeObserver?.disconnect();
+                        resizeObserver = null;
+                    }
+                }
             }
+        };
+
+        if (props.autoCut) {
+            resizeObserver = new ResizeObserver(handleResize);
```

Comment:
What if content is just super long block of text without images? It will never resize so it won't collapse.
Need to force autocut first and if not cut- schedule `ResizeObserver`

Reply:
Lemmy test this.

Reply:
Seems to work just fine as is.

Reply:
Hm, unless there is autocut implemented somewhere else - I don't see how it may be working

Reply:
Ok, I'll rewrite this just in case.

Reply:
Fixed this and some other corner cases (mainly the interaction with the low rating blur).